libs/initScripts/local.js

```diff
@@ -0,0 +1,124 @@
+const initAudiusLibs = require('../examples/initAudiusLibs')
+const { distributeTokens } = require('./helpers/distributeTokens')
+const { setServiceVersion } = require('./helpers/version')
+const { registerLocalService, queryLocalServices, getStakingParameters } = require('./helpers/spRegistration')
+const { deregisterLocalService } = require('./helpers/spRegistration')
+const { getClaimInfo, fundNewClaim } = require('./helpers/claim')
+
+const serviceTypeList = ['discovery-provider', 'creator-node', 'content-service']
+const spDiscProvType = serviceTypeList[0]
+const spCreatorNodeType = serviceTypeList[1]
+const discProvEndpoint1 = 'http://docker.for.mac.localhost:5000'
+const creatorNodeEndpoint1 = 'http://docker.for.mac.localhost:4000'
+const creatorNodeEndpoint2 = 'http://docker.for.mac.localhost:4010'
+const versionStr = '0.1.0'
+
+const throwArgError = () => {
+  throw new Error(`missing argument - format: node examples/initiateStakingOperations.js [
+    distribute,
+    fundclaim,
+    getclaim,
+    stakeinfo,
+    setversion,
+    register-sps,
+    deregister-sps,
+    query-sps,
+    init-all
+  ]`)
+}
+
+let args = process.argv
+if (args.length < 3) {
+  throwArgError()
+}
+
+const run = async () => {
+  try {
+    let audiusLibs = await initAudiusLibs(true)
+    let ethWeb3 = audiusLibs.ethWeb3Manager.getWeb3()
+    const ethAccounts = await ethWeb3.eth.getAccounts()
+
+    switch (args[2]) {
+      case 'distribute':
+        console.log('distribute')
+        await distributeTokens(audiusLibs)
+        break
+
+      case 'fundclaim':
+        console.log('fundclaim')
+        await fundNewClaim(audiusLibs)
+        break
+
+      case 'getclaim':
+        console.log('getclaim')
+        await getClaimInfo(audiusLibs)
+        break
+
+      case 'stakeinfo':
+        console.log('stakeinfo')
+        await getStakingParameters(audiusLibs)
+        break
+
+      case 'setversion':
+        await _initAllVersions(audiusLibs)
+        break
+
+      case 'register-sps':
+        await _registerAllSPs(audiusLibs, ethAccounts)
+        break
+
+      case 'deregister-sps':
+        await _deregisterAllSPs(audiusLibs, ethAccounts)
+        break
+
+      case 'query-sps':
+        await queryLocalServices(audiusLibs, serviceTypeList)
+        break
+
+      case 'init-all':
+        await _initializeLocalEnvironment(audiusLibs, ethAccounts)
+        break
+      default:
+        throwArgError()
+    }
+
+    process.exit(0)
+  } catch (e) {
+    throw e
+  }
+}
+
+run()
+
+const _initializeLocalEnvironment = async (audiusLibs, ethAccounts) => {
+  await distributeTokens(audiusLibs)
+  await _initAllVersions(audiusLibs)
```

Comment:
jw how is this function naming convention intended - are _functions different than non "_" functions? both look like they serve command line arguments, it doesn't seem like some are internal vs others?

Reply:
@hareeshnagaraj functions intended to be private are generally prefixed with _. because js doesn't have actual private function scoping this is a convention. in this case the cli is like the external api. anything not directly callable is "private"